[yorklim] iP

yorklim opened the pull request:
# NotDuke

>"Your mind is for having ideas, not holding them." – David Aleen ([source](https://dansilvestre.com/productivity-quotes))

NotDuke frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

All you need to do is,

1. download it from [here](https://github.com/yorklim/ip/releases).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you :wink:

And its **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

yorklim committed:
Added Level 0. Rename, Greet, Exit

yorklim committed:
Added Level 1. Echo

yorklim committed:
Added Level 2. Add, List

yorklim committed:
Added Level 3. Mark as Done

yorklim committed:
Added Level 4. ToDos, Events, Deadlines

yorklim committed:
Added Level 5. Handle Errors

yorklim committed:
Added Level 6. Delete

yorklim committed:
Add Level 7. Save

yorklim committed:
Add Level 7. Save

yorklim committed:
add Level 7. Save

yorklim committed:
Merge branch 'branch-Level-7'

yorklim committed:
Merge branch 'branch-Level-7'

yorklim committed:
Add Level 8. Dates and Times

yorklim committed:
Add Level 8. Dates and Times

yorklim committed:
Merge branch 'branch-Level-8'

yorklim committed:
Add more OOP

yorklim committed:
Move files to packages

yorklim committed:
Add gradle support

yorklim committed:
Add JUnit tests for Task classes

yorklim committed:
bug fix: Add line seperator after list command

yorklim committed:
Add JavaDoc

yorklim committed:
Remerge branch-Level-7

yorklim committed:
Remerge branch-Level-8

yorklim committed:
Change Coding Standard

yorklim committed:
Add Level 9. Find

yorklim committed:
Merge branch 'branch-Level-9'

yorklim committed:
Fix branch-A-CodingStandard merge conflict

yorklim committed:
Fix branch-A-JavaDoc merge conflict

yorklim committed:
Add javadoc to Task methods

yorklim committed:
Remove leftover merge conflict tags

yorklim committed:
Update Readme.md and Add Checkstyle

yorklim committed:
Update Readme.md

ongzhili commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,17 @@
+import duke.storage.Storage;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+public class ByeCommand extends Command {
```
I like that the imports are sorted alphabetically. However, would it be better to leave a line after line 6 to improve readability?

ongzhili commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,50 @@
+    public static Command parse(String fullCommand) throws DukeInvalidCommand {
+        String[] input = fullCommand.split(" ", 2);
+        String command = input[0];
+        if (command.equals("bye")) {
```
I like that the if else statements are formatted nicely, but I would also like to suggest switch case as a slightly less verbose alternative.

ongzhili commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,155 @@
+     * Returns a string to be printed which contains all the String representation of the tasks in the TaskList.
+     * @return a string representation of all the task in the TaskList.
+     */
+    public String list() {
```
Since there is a list interface already in Java, should this method name be changed to prevent confusion?

ongzhili commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,155 @@
+ * A List that is used to contain the Tasks inputted by the user.
+ */
+public class TaskList {
+    private ArrayList<Task> list;
```
Since there is a list interface already in Java, should this variable name be changed to prevent confusion?

ongzhili commented on src/main/java/duke/command/CheckCommand.java:
```diff
@@ -0,0 +1,31 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
```
I like the separation of import statements. Common mistake people leave out.

ongzhili commented on src/main/java/duke/task/Events.java:
```diff
@@ -0,0 +1,68 @@
+
+    @Override
+    public String happenOn(LocalDate date) {
+        if ((date.isAfter(from.toLocalDate()) && date.isBefore(to.toLocalDate()))
+                || date.isEqual(from.toLocalDate()) || date.isEqual(to.toLocalDate())) {
```
I like the clean like break.

ongzhili commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,108 @@
+
+    public String taskInfo() {
+        String output = "";
+        if (isDone) {
+            output += "[X]";
+        } else {
+            output += "[ ]";
+        }
+        return output + " " + name;
```
Could this be shortened to the following?
```suggestion
        if (isDone) {
            return "[X] " + name;
        } else {
            return "[ ] " + name;
        }
```

ongzhili reviewed (commented):
Looks good 👍 .
Some small changes suggested.

yorklim committed:
Add Level-10 GUI

yorklim committed:
Merge branch 'JavaFX'

yorklim committed:
Add Assertions to code

There are a couple of methods that if ran without any bugs would be
able to ignore cases where list size is less than 0.

Adding assertions to those method would ensure that "illegal" values
of list size can be identified and would stop the program from
continuing to run and allow fixing of bugs

yorklim committed:
Improve code quality

Improved the code quality of the methods to allow for better
readability and make it easier to understand for future developers
that would like to work on this code.

yorklim committed:
Merge pull request #1 from yorklim/branch-A-Assertions

Add Assertions to code

yorklim committed:
Merge branch 'master' into branch-A-CodeQuality

yorklim committed:
Fix merge conflicts

yorklim committed:
Merge branch 'branch-A-CodeQuality' of github.com:yorklim/ip into branch-A-CodeQuality

yorklim committed:
Merge pull request #2 from yorklim/branch-A-CodeQuality

Improve code quality

yorklim committed:
Set up CI

yorklim committed:
Add reminders and fix bugs

Added command remind which allows users to check what are occurring
or happening within 7 days. On startup, the reminder will also be
shown.

Fixed bugs regarding exceptions not being handled when only the command
is given eg. "todo" for commands that require arguments.

yorklim committed:
Merge pull request #3 from yorklim/branch-B-Reminders

Add reminders and fix bugs

yorklim committed:
Change references of Duke to NotDuke

yorklim committed:
Add Ui.png

yorklim committed:
Update Readme.md

yorklim committed:
Update Readme.md

yorklim committed:
Fix warning messages at startup

yorklim committed:
Add Javadoc to new methods in Task and TaskList
